.storybook/webpack.config.js

```diff
@@ -96,69 +94,66 @@ if (USER_ID && USER_ACCESS_TOKEN) {
   )
 }
 
-// A mix of  the base from Emission's webpack setup, and the simple config for
-// storybooks: https://storybook.js.org/configurations/custom-webpack-config/
-
 console.log("\n[Reaction] Booting...\n")
 
 /**
- * @type {webpack.Configuration}
+ * Booting in full-control mode: https://storybook.js.org/docs/configurations/custom-webpack-config/#full-control-mode-default
  */
-module.exports = {
-  devtool: WEBPACK_DEVTOOL,
-  devServer: {
+module.exports = async ({ config, mode }) => {
+  config.mode = mode.toLowerCase()
+  config.devtool = WEBPACK_DEVTOOL
+  config.devServer = {
     overlay: {
       warnings: true,
       errors: true,
     },
     stats: "errors-only",
-  },
-
-  resolve: {
+  }
+  config.resolve = {
     extensions: [".mjs", ".js", ".jsx", ".ts", ".tsx"],
     alias: {
       sharify: sharifyPath.replace(/\.js$/, ""),
       "styled-components": path.resolve("./node_modules/styled-components"),
     },
-  },
-  module: {
-    rules: [
-      {
-        test: /\.graphql$/,
-        include: [/data/],
-        exclude: [/node_modules/],
-        use: [
-          {
-            loader: "raw-loader",
-          },
-        ],
-      },
-      {
-        test: /\.tsx?$/,
-        include: [/src/],
-        exclude: [/node_modules/, new RegExp(package.jest.testRegex)],
-        use: [
-          {
-            loader: "cache-loader",
-            options: {
-              cacheDirectory: path.join(cacheDirectory),
-            },
+  }
+  config.plugins = plugins
```

Comment:
I notice that the below `rules` array uses push, are we sure that `plugins` does not use this as well?

Reply:
We're replacing it outright in this case, but that's a good catch.